docs/source/data-types/timeuuid.md

```diff
@@ -27,4 +27,47 @@ if let Some(rows) = session.query("SELECT a FROM keyspace.table", &[]).await?.ro
 }
 # Ok(())
 # }
-```
\ No newline at end of file
+```
+
+## Creating your own Timeuuid
+
+To create your own `Timeuuid` objects from timestamp-based `uuid` v1, you need to enable the feature `v1` of `uuid` crate using:
+
+```shell
+cargo add uuid -F v1
+```
+
+and now you're gonna be able to use the `uuid::v1` features: 
+
+```rust
+# extern crate uuid;
+# extern crate scylla;
+# use scylla::Session;
+# use std::error::Error;
+# use std::str::FromStr;
+use scylla::IntoTypedRows;
+use scylla::frame::value::CqlTimeuuid;
+use uuid::Uuid;
+# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
+
+// Tip: you can use random stable numbers or your MAC Address
+let node_id = [0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC];
+
+// Build your Timeuuid with the current timestamp
+let to_insert = CqlTimeuuid::from(Uuid::now_v1(&node_id));
+
+session
+    .query("INSERT INTO keyspace.table (a) VALUES(?)", (to_insert,))  
+    .await?;
+
+// Read timeuuid from the table
+if let Some(rows) = session.query("SELECT a FROM keyspace.table", &[]).await?.rows {
+    for row in rows.into_typed::<(CqlTimeuuid,)>() {
+        let (timeuuid_value,): (CqlTimeuuid,) = row?;
+    }
+}
```

Comment:
Don't use raw `rows` field on `QueryResult`; prefer `rows_typed::<(CqlTimeuuid,)>()` method:

```suggestion
let mut iter = session.query("SELECT a FROM keyspace.table", &[]).await?.rows_typed::<(CqlTimeuuid,)>();
while let Some((timeuuid_value,)) = iter.next().transpose()? {
  // do something with timeuuid_value
}
```

Reply:
I'll update, but can you explain in details why I shouldn't use it?

Reply:
Fixed! Ready to merge.

Reply:
`rows` make you interact with an untyped, materialised vec of columns. At the present state of the driver, the main drawback is reduced type safety; as you call `into_typed` anyway, this negative aspect is mitigated.
Another reason for avoiding using `rows` ATM, which is relevant here, is the ongoing deserialization refactor. We ditch the materialized vec of `CqlValue` at all to reduce unnecessary allocations. The new deserialization framework is lazy, which means that the values are going to be deserialized on-the-fly straight to the end type, based on the type provided in the iterator (`rows_typed::<T>`).
With the new framework having landed in the driver, `rows` will be removed at all (although their capabilities can be emulated for the users that *really* need them).

See https://github.com/scylladb/scylla-rust-driver/pull/955